lms/models/solutions.py

```diff
@@ -6,13 +6,50 @@
 
 from flask_babel import gettext as _
 
-from lms.lmsdb.models import Solution, SolutionFile
+from lms.lmsdb.models import Comment, Solution, SolutionFile, User
 from lms.lmstests.public.general import tasks as general_tasks
 from lms.lmstests.public.identical_tests import tasks as identical_tests_tasks
 from lms.lmsweb import config, routes
 from lms.models import notifications
 
 
+def send_notification_after_check(user: User, solution: Solution) -> bool:
+    is_checked = solution.is_checked
+    if is_checked:
+        if solution.solver == user:
+            msg = _(
+                '%(solver)s הגיב לך על בדיקת תרגיל "%(subject)s".',
+                solver=solution.solver.fullname,
+                subject=solution.exercise.subject,
+            )
+            addressee = solution.checker
+        else:  # solution.checker == user
+            msg = _(
+                '%(checker)s הגיב לך על תרגיל "%(subject)s".',
+                checker=solution.checker.fullname,
+                subject=solution.exercise.subject,
+            )
+            addressee = solution.solver
+        if (
+            not solution.comments
+            or (
+                solution.comments
+                and not solution.comments.order_by(
+                    Comment.timestamp.desc(),
+                )[0].commenter == user
+            )
+        ):
+            notifications.send(
```

Comment:
I can't understand the flow. Why we send a notification if `not solution.comments`?

Reply:
First of all we check that the solution has checked.
Then there are two possible situations. There is a possibility that the checker didn't comment at all. And the second one is to check that the last commenter is the checker (if the solver comments) or the solver (if the checker comments).